diff --git a/setup.py b/setup.py
index 0decaf66ed6..fb4158642de 100644
--- a/setup.py
+++ b/setup.py
@@ -396,6 +396,8 @@ if BUILD_PYTHON_ONLY:
 
 VERBOSE_SCRIPT = str2bool(os.getenv("VERBOSE", "1"))
 RUN_BUILD_DEPS = True
+if os.getenv("SKIP_BUILD_DEPS", "").lower() in {"1", "true", "yes", "on"}:
+    RUN_BUILD_DEPS = False
 # see if the user passed a quiet flag to setup.py arguments and respect
 # that in our parts of the build
 EMIT_BUILD_WARNING = False
