diff --git a/src/cpp/include/lemon/backends/llamacpp/llamacpp.h b/src/cpp/include/lemon/backends/llamacpp/llamacpp.h
index 12a9a506a..c0ef44383 100644
--- a/src/cpp/include/lemon/backends/llamacpp/llamacpp.h
+++ b/src/cpp/include/lemon/backends/llamacpp/llamacpp.h
@@ -41,7 +41,6 @@ inline const BackendDescriptor descriptor = {
          "Logit scale for reranking relevance scores", "Llama.cpp Reranking Options"},
     },
     /*support*/ {
-        {"system", {"linux"}, {{"cpu", {"x86_64", "arm64"}}}, "x86_64/ARM64 CPU, GPU"},
         {"metal", {"macos"}, {{"metal", {}}}, "Apple Silicon GPU"},
         {"cuda", {"windows", "linux"},
          {{"nvidia_gpu", {"sm_75", "sm_80", "sm_86", "sm_89", "sm_90", "sm_100", "sm_120", "sm_121"}}}, "NVIDIA GPUs (Turing or newer)**"},
